### ISSUE TITLE
Error and loading states

### ISSUE THREAD

**Issue by andrew-c-tran** [drud/style.drud.com#4]
The UI will regularly fall into a state where actions are happening and an indicator that work is being done needs to be shown. A loading spinner or other type of indicator is needed. 

The UI will also occasionally fall into a state where an action has failed, or either the UI or the CLI has experienced some sort of error. Defining error prompts and other indicators on how to display the correct error information is needed.

**Comment by andrew-c-tran** [comment]
Initial proposal for loading state:

grey overlay with a loading spinner and current state to the right. in card view, this will overlay the card processing things, in modals it will overlay the modal.

example:
![screen shot 2017-12-12 at 10 24 18 am](https://user-images.githubusercontent.com/8960363/33898873-e62b4c9e-df26-11e7-9a3a-df9c7fccb08e.png)

**Comment by rickmanelius** [comment]
Crap I missed this earlier re:full view versus only covering an individual card. Disregard that comment. As for the elements/colors, I think we need to definitely address color contrast because things like the "Creating ddev-router" wash out a bit. That said, it's easy to note the deficiency and hard to address the correction because I know we may need additional support from a design resource to make more definitive recommendations.

TL;DR I like the strategy but work needs to be done on the design aspects of colors/sizing/placement.

**Comment by cyberswat** [comment]
We really need to commit to getting the efforts from the style guide wrapped into this kind of work.  I understand that it's a bit of a pain because the people involved are a bit isolated, but it feels like something we should fix.

**Comment by andrew-c-tran** [comment]
We should have the error and loading states today correct? We can
definitely pull those in

**Comment by rickmanelius** [comment]
Related to https://github.com/drud/ddev-ui/issues/8
